src/pages/ReportDetailsPage.tsx

```diff
@@ -395,6 +625,23 @@ function ReportDetailsPage({policies, report, session, personalDetails}: ReportD
                     confirmText={translate('common.leave')}
                     cancelText={translate('common.cancel')}
                 />
+                <ConfirmModal
+                    title={caseID === CASES.DEFAULT ? translate('task.deleteTask') : translate('iou.deleteExpense')}
+                    isVisible={isDeleteModalVisible}
+                    onConfirm={deleteTransaction}
+                    onCancel={() => setIsDeleteModalVisible(false)}
+                    onModalHide={() => {
```

Comment:
There is a race condition between deleting a transaction and navigating back. This results in brief moment where transaction page is still open but there is no data(showing `Deleted transaction`. This is primarily noticeable in slower devices. https://github.com/Expensify/App/issues/45576